assets/src/js/Components/ArchivePicker/ArchivePicker.js

```diff
@@ -356,6 +364,23 @@ export default function ArchivePicker({view = ADMIN_VIEW}) {
     });
   };
 
+  const setCustomFeaturedImage = async imgId => {
+    await wp.data.dispatch('core/editor').editPost({featured_media: imgId});
+    dispatch({type: ACTIONS.CLOSE_SIDEBAR});
+    document.querySelector('.media-modal-close .media-modal-icon').click();
+  };
+
+  // Prevent multiple instances of the Media Upload Modal.
+  const mediaModalPopup = document.querySelectorAll('.media-modal');
+
+  useEffect(() => {
+    if (mediaModalPopup && mediaModalPopup.length > 1) {
+      const modals = Array.from(mediaModalPopup);
```

Comment:
Thinking loud.. is `[...mediaModalPopup]` a better approach?

Reply:
`mediaModalPopup` is a NodeList we are converting to an array, so hence the approach above, I think 🤔